src/main/java/study/step3/CarRacing.java

```diff
@@ -0,0 +1,40 @@
+package study.step3;
+
+import study.step3.dto.Input;
+import study.step3.move.Move;
+import study.step3.move.RandomlyMove;
+
+import java.util.*;
+
+public class CarRacing {
+
+
+    public static void main(String[] args) {
+
+        InputView inputView = new InputView();
+        ResultView resultView = new ResultView();
+
+        resultView.print(racing(inputView.enter(), new RandomlyMove()));
+
+    }
+
+    public static String[][] racing(Input input, Move move) {
+        int carNum = input.getCarNum();
+        int tryNum = input.getTryNum();
+        String[][] positionArr = new String[tryNum][carNum];
+
+        for (String[] row : positionArr) {
+            Arrays.fill(row, "-");
+        }
+
+        for (int i = 1; i < tryNum; i++) {
+            for (int j = 0; j < carNum; j++) {
+                if (move.isSuccess()){
+                    positionArr[i][j] = positionArr[i - 1][j] + "-";
+                }
+            }
+        }
+
+        return positionArr;
+    }
```

Comment:
```suggestion
    public static void main(String[] args) {
        InputView inputView = new InputView();
        ResultView resultView = new ResultView();
        Input request = inputView.enter();

        RacingGame game = new Racing(request.getCarNum(), new RandomlyMove());

        for (int i = 0; i < request.getTryNum(); i++) {
            game.play();
            List<Car> cars = game.getCars();
            resultView.printCars(cars);
        }
    }
```

기존 코드는 절차지향성이 강한 것 같아요
클래스를 분리해보고 각 클래스들에게 적절한 책임을 가질 수 있도록 코드를 작성해 볼 수 없을까요?

Reply:
현재 도메인 클래스가 없기 때문에 테스트 코드도 없는 것 같아요
위 처럼 코드를 작성했다면 Car 클래스에 대한 테스트 코드도 만들 수 있어보입니다.